src/test/parse-fail/obsolete-proc.rs

```diff
@@ -10,8 +10,8 @@
 
 // Test that we generate obsolete syntax errors around usages of `proc`.
 
-fn foo(p: proc()) { } //~ ERROR obsolete syntax: the `proc` type
+fn foo(p: proc()) { } //~ ERROR: the `proc` type isn't used for the moment
```

Comment:
I removed `ProcExpr`, `ProcType` and `parse_proc_type`. I'll run `make check-stage2-pfail` tomorrow.